File: PythonNBA/scripts/_compare.py
# scripts/_compare.py
import json

def main():
    # Compare the original backup vs recalculated history
    with open("data/history_pre_recalc_2026-03-15T15-00-39.json", "r") as f:
        orig = json.load(f)
    with open("data/history.json", "r") as f:
        curr = json.load(f)

    orig_runs = orig.get("runs", [])
    curr_runs = curr.get("runs", [])

    print("=== Original vs Recalculated ===")
    print(f"Runs: {len(orig_runs)} vs {len(curr_runs)}\n")

    # Compare weights
    print("Original weights:", json.dumps(orig.get("weights", {}))[:200])
    print("Current weights:", json.dumps(curr.get("weights", {}))[:200])
    print()

    # Count picks per run
    orig_w = orig_l = curr_w = curr_l = 0
    orig_picks = curr_picks = 0

    for i in range(len(orig_runs)):
        o_run = orig_runs[i] if i < len(orig_runs) else None
        c_run = curr_runs[i] if i < len(curr_runs) else None
        if not o_run or not c_run:
            continue

        o_w = o_l = c_w = c_l = o_picks_count = c_picks_count = 0

        for g in o_run.get("games", []):
            s_pick = g.get("sPick")
            s_conf = g.get("sConf")
            if s_pick and s_pick != "PASS" and s_conf in ("high", "elite"):
                o_picks_count += 1
                if g.get("sResult") == "WIN":
                    o_w += 1
                if g.get("sResult") == "LOSS":
                    o_l += 1

        for g in c_run.get("games", []):
            s_pick = g.get("sPick")
            s_conf = g.get("sConf")
            if s_pick and s_pick != "PASS" and s_conf in ("high", "elite"):
                c_picks_count += 1
                if g.get("sResult") == "WIN":
                    c_w += 1
                if g.get("sResult") == "LOSS":
                    c_l += 1

        orig_w += o_w
        orig_l += o_l
        orig_picks += o_picks_count
        curr_w += c_w
        curr_l += c_l
        curr_picks += c_picks_count

        if o_picks_count != c_picks_count or o_w != c_w:
            print(f"{o_run.get('dateDisplay')}: orig {o_w}-{o_l} ({o_picks_count} picks) -> recalc {c_w}-{c_l} ({c_picks_count} picks)  [{o_picks_count - c_picks_count} picks lost]")

    orig_total = orig_w + orig_l
    curr_total = curr_w + curr_l
    orig_pct = f"{orig_w / orig_total * 100:.1f}" if orig_total > 0 else "0.0"
    curr_pct = f"{curr_w / curr_total * 100:.1f}" if curr_total > 0 else "0.0"

    print(f"\nOriginal total: {orig_w}-{orig_l} ({orig_picks} picks) {orig_pct}%")
    print(f"Recalc total:   {curr_w}-{curr_l} ({curr_picks} picks) {curr_pct}%")
    print(f"Lost: {orig_picks - curr_picks} picks, {orig_w - curr_w} wins, {orig_l - curr_l} losses")

    # Check: are the lost picks from high-conf or from a threshold change?
    print("\n=== Checking what changed ===")
    # Compare probHigh thresholds
    for i in range(len(orig_runs)):
        if i >= len(curr_runs):
            break
        o_w_used = orig_runs[i].get("weightsUsed") or {}
        c_w_used = curr_runs[i].get("weightsUsed") or {}
        if not o_w_used or not c_w_used:
            continue
        if o_w_used.get("probHigh") != c_w_used.get("probHigh") or \
           o_w_used.get("probElite") != c_w_used.get("probElite"):
            print(f"{orig_runs[i].get('dateDisplay')}: probHigh {o_w_used.get('probHigh')}->{c_w_used.get('probHigh')}  probElite {o_w_used.get('probElite')}->{c_w_used.get('probElite')}")

if __name__ == "__main__":
    main()

File: jsNBA/scripts/_compare.mjs
import fs from 'fs';

// Compare the original backup vs recalculated history
const orig = JSON.parse(fs.readFileSync('data/history_pre_recalc_2026-03-15T15-00-39.json', 'utf8'));
const curr = JSON.parse(fs.readFileSync('data/history.json', 'utf8'));

const origRuns = orig.runs || [];
const currRuns = curr.runs || [];

console.log('=== Original vs Recalculated ===');
console.log(`Runs: ${origRuns.length} vs ${currRuns.length}\n`);

// Compare weights
console.log('Original weights:', JSON.stringify(orig.weights).slice(0, 200));
console.log('Current weights:', JSON.stringify(curr.weights).slice(0, 200));
console.log();

// Count picks per run
let origW = 0, origL = 0, currW = 0, currL = 0;
let origPicks = 0, currPicks = 0;

for (let i = 0; i < origRuns.length; i++) {
  const oRun = origRuns[i];
  const cRun = currRuns[i];
  if (!oRun || !cRun) continue;

  let oW = 0, oL = 0, cW = 0, cL = 0, oPicks = 0, cPicks = 0;

  for (const g of oRun.games || []) {
    if (g.sPick && g.sPick !== 'PASS' && (g.sConf === 'high' || g.sConf === 'elite')) {
      oPicks++;
      if (g.sResult === 'WIN') oW++;
      if (g.sResult === 'LOSS') oL++;
    }
  }

  for (const g of cRun.games || []) {
    if (g.sPick && g.sPick !== 'PASS' && (g.sConf === 'high' || g.sConf === 'elite')) {
      cPicks++;
      if (g.sResult === 'WIN') cW++;
      if (g.sResult === 'LOSS') cL++;
    }
  }

  origW += oW; origL += oL; origPicks += oPicks;
  currW += cW; currL += cL; currPicks += cPicks;

  if (oPicks !== cPicks || oW !== cW) {
    console.log(`${oRun.dateDisplay}: orig ${oW}-${oL} (${oPicks} picks) → recalc ${cW}-${cL} (${cPicks} picks)  [${oPicks - cPicks} picks lost]`);
  }
}

console.log(`\nOriginal total: ${origW}-${origL} (${origPicks} picks) ${(origW/(origW+origL)*100).toFixed(1)}%`);
console.log(`Recalc total:   ${currW}-${currL} (${currPicks} picks) ${(currW/(currW+currL)*100).toFixed(1)}%`);
console.log(`Lost: ${origPicks - currPicks} picks, ${origW - currW} wins, ${origL - currL} losses`);

// Check: are the lost picks from high-conf or from a threshold change?
console.log('\n=== Checking what changed ===');
// Compare probHigh thresholds
for (let i = 0; i < origRuns.length; i++) {
  const oW = origRuns[i]?.weightsUsed;
  const cW = currRuns[i]?.weightsUsed;
  if (!oW || !cW) continue;
  if (oW.probHigh !== cW.probHigh || oW.probElite !== cW.probElite) {
    console.log(`${origRuns[i].dateDisplay}: probHigh ${oW.probHigh}→${cW.probHigh}  probElite ${oW.probElite}→${cW.probElite}`);
  }
}
